### ISSUE TITLE
Avoid recursion for next.Run

### ISSUE THREAD

**Issue by state303** [state303/grace#8]
Currently each call for next represents depth. This is not good.
We need some other way to resolve this. It needs to be fixed along with #7

**Comment by state303** [comment]
```go
type Task interface {
	Run(ctx context.Context) error
	Then(that Task) Task
	Step() Step // as internal function that runs under specific task 
        Next() Task // also required to properly chain via loop
}
```

**Comment by state303** [comment]
Fixed in PR #10
